src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -132,16 +132,7 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMeth
                 {
                     if (!segment.Value.IsConstant && segment.Value.Reference.Name == "nextLink")
                     {
-                        if (segment.IsRaw)
-                        {
-                            // Artificial nextLink needs additional logic for relative versus absolute links
-                            WritePathSegment(writer, uri, segment, "AppendRawNextLink");
-                        }
-                        else
-                        {
-                            // Natural nextLink parameters need to use a different method to parse path and query elements
-                            WritePathSegment(writer, uri, segment, "AppendRaw");
-                        }
+                        WritePathSegment(writer, uri, segment, "AppendRawNextLink");
```

Comment:
Just out of curiosity - why was this distinction between `IsRaw` ever important? Does it also make sense for `AppendRaw` to detect when the segment being appended is a superset of the target uri rather than special casing it in `AppendRawNextLink`?

Reply:
I think we tried to reduce the number of cases where we run a slightly more expensive `AppendRawNextLink`. Not sure if it was worth it.